### PR TITLE
fix kalman filter

### PR DESCRIPTION
**Pull request by 2141Robotics**

### CODE REVIEW

**Comment by Mineinjava** [comment]
Appears to work. Will merge when confidence is higher

File: quail/src/main/java/com/mineinjava/quail/localization/KalmanFilterLocalizer.java [hunk 1]
```diff
@@ -48,8 +48,14 @@ public Vec2d update(Vec2d observedPose, Vec2d velocity, double poseEstimateLaten
     Vec2d kinematicsPoseEstimate = this.poseEstimate.add(velocity.scale(this.looptime));
     // update the pose estimate with a weighted average of the vision and kinematics pose estimates
     w = Util.clamp(w, 0, 1); // make sure w is between 0 and 1 (inclusive)
-    this.poseEstimate =
-        visionPoseEstimate.scale(w).add(kinematicsPoseEstimate.scale(1 - w)).scale(0.5);
+    System.out.println(
+        "k "
```
Comment (Mineinjava) [inline quail/src/main/java/com/mineinjava/quail/localization/KalmanFilterLocalizer.java:52]: Remove print

**Review by Mineinjava — CHANGES_REQUESTED** [review]
Looks good, just remove the print

**Comment by Bernstern** [comment]
Instead of removing the print I'd add maybe a debug flag to the constructor or some static constant shared across quail for verbose logging 🤷 

I'd also move the `w` param to the constructor since I'm assuming it won't change between calls.

**Comment by Mineinjava** [comment]
> maybe a debug flag to the constructor or some static constant shared across quail for verbose logging 🤷

This should probably be done properly with logging or some sort of making variables gettable for smart dashboard

**Comment by Mineinjava** [comment]
> I'd also move the w param to the constructor since I'm assuming it won't change between calls.

W will change between calls as vision noise increases with distance and other variables

**Review by Mineinjava — APPROVED** [review]

**Pull request merged**